socorro/external/es/connection_context.py

```diff
@@ -93,24 +98,49 @@ def __init__(self, config):
     def connection(self, name=None, timeout=None):
         """Returns an instance of elasticsearch-py's Elasticsearch class as
         encapsulated by the Connection class above.
+
         Documentation: http://elasticsearch-py.readthedocs.org
+
         """
         if timeout is None:
             timeout = self.config.elasticsearch_timeout
 
+        # Separate the HTTP from HTTPS urls so we know if they're all one or
+        # the other because we don't want a mix
+        http = [url for url in self.config.elasticsearch_urls if not url.startswith('https')]
+        https = [url for url in self.config.elasticsearch_urls if url.startswith('https')]
+
+        if http and https:
+            raise ElasticsearchConfigurationError(
+                'elasticsearch_urls must either be all http or all https--you cannot mix them.'
+            )
+
+        if https:
+            # If we have https urls, then we want to add these additional
+            # connection arguments to ensure secure connections
+            additional_kwargs = {
+                'verify_certs': True,
+                'use_ssl': True
+            }
+        else:
+            additional_kwargs = {}
+
         return Connection(
             self.config,
             elasticsearch.Elasticsearch(
                 hosts=self.config.elasticsearch_urls,
                 timeout=timeout,
-                connection_class=elasticsearch.connection.RequestsHttpConnection
+                connection_class=elasticsearch.connection.RequestsHttpConnection,
```

Comment:
Does anything in elasticsearch itself verify that the URLs have protocols and they're all either http or https?

Reply:
The library has this code:

https://github.com/elastic/elasticsearch-py/blob/master/elasticsearch/connection/http_requests.py#L45

So you pass in a host and a port and it uses `use_ssl` to determine whether to use http or https in the url.

The socorro elasticsearch connection context pass in the value of elasticsearch_urls in to create the Elasticsearch client instance. In the `__init__` for that, it normalizes the hosts:

https://github.com/elastic/elasticsearch-py/blob/cdb04dfa011600facdfdd8d2eacd7dded4404413/elasticsearch/client/__init__.py

That normalization process assumes everything is http unless it has the https protocol in which case it adds `use_ssl=True`. So... now that I'm looking at that, we can probably drop our code for adding `use_ssl`. It's even possible that we could set `verify_certs=True` regardless of http or https.

I'll check that out tomorrow. If that's true, then this PR gets much much shorter.